### ISSUE TITLE
When printing an `AbstractOperation`, the tree should be indented for readability (and without newline separation)

### ISSUE THREAD

**Issue by ali-ramadhan** [CliMA/Oceananigans.jl#893]
Probably unintentional but instead of

```
julia> T*S
BinaryOperation at (Cell, Cell, Cell)
├── grid: RegularCartesianGrid{Float64,Periodic,Periodic,Bounded,OffsetArrays.OffsetArray{Float64,1,StepRangeLen{Float64,Base.TwicePrecision{Float64},Base.TwicePrecision{Float64}}}}
│   ├── size: (16, 16, 16)
│   └── domain: x ∈ [0.0, 1.0], y ∈ [0.0, 1.0], z ∈ [-1.0, 0.0625]
└── tree: 

* at (Cell, Cell, Cell) via identity
├── OffsetArray{Float64, 3, Array{Float64,3}}
└── OffsetArray{Float64, 3, Array{Float64,3}}
```

this might be more readable

```
julia> T*S
BinaryOperation at (Cell, Cell, Cell)
├── grid: RegularCartesianGrid{Float64,Periodic,Periodic,Bounded,OffsetArrays.OffsetArray{Float64,1,StepRangeLen{Float64,Base.TwicePrecision{Float64},Base.TwicePrecision{Float64}}}}
│   ├── size: (16, 16, 16)
│   └── domain: x ∈ [0.0, 1.0], y ∈ [0.0, 1.0], z ∈ [-1.0, 0.0625]
└── tree: 
    * at (Cell, Cell, Cell) via identity
    ├── OffsetArray{Float64, 3, Array{Float64,3}}
    └── OffsetArray{Float64, 3, Array{Float64,3}}
```

See #892 for a minimal working example

**Comment by glwagner** [comment]
Note also that `domain` info from `show` is incorrect

**Comment by ali-ramadhan** [comment]
Ah I thought I fixed that as item 6 in PR #823 but looks like I only fixed it for `Base.show(io, ::RegularCartesianGrid)` and not for `show_domain(grid)`.
